x-pack/legacy/plugins/siem/server/lib/detection_engine/signals/signal_rule_alert_type.ts

```diff
@@ -209,13 +211,29 @@ export const signalRulesAlertType = ({
             `[+] Initial search call of signal rule name: "${name}", id: "${alertId}", rule_id: "${ruleId}"`
           );
           const noReIndexResult = await services.callCluster('search', noReIndex);
-          if (noReIndexResult.hits.total.value !== 0) {
+          const newSignalsCount = noReIndexResult.hits.total.value;
+
+          if (newSignalsCount !== 0) {
+            const inputIndexes = inputIndex.join(', ');
+
+            if (throttle && throttle !== 'no_actions') {
+              const alertInstance = services.alertInstanceFactory(ruleId!);
```

Comment:
I think this should be using the `alertId` in place of the `ruleId`. This will also get rid of the bang.